### PR TITLE
Create tests for suggestions & implement loading & error state handling

### LINKED ISSUE #83: Show loader when retrieving suggestions
- [ ] Show loader when retrieving suggestions
- [ ] Show loader when retrieving data for suggestion on suggestion detail page
- [ ] Remove // TODO on both pages

### LINKED ISSUE #107: Suggestions error handling
- [ ] Show error when failed to retrieve suggestions
- [ ] Show error when failed to retrieve data on suggestion detail page
- [ ] See //TODO in [suggestion] & pages/suggestions/index

### LINKED ISSUE #156: Write suggestion card and details tests

### PR DESCRIPTION
**Pull request by fkoulen**
Closes #83 
Closes #107 
Closes #156

### CODE REVIEW

File: cypress/e2e/suggestion.cy.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const suggestionId = Cypress.env("CYPRESS_SUGGESTION_ID")
+
+describe("suggestion detail page", () => {
```
Comment (fkoulen) [inline cypress/e2e/suggestion.cy.ts:3]: This misses tests for entering what instrument you're playing.

File: cypress/e2e/suggestions.cy.ts
Comment (fkoulen) [inline cypress/e2e/suggestions.cy.ts:1]: There are no tests here for edge cases such as a failing requests, the loading spinner or when there are no requests. I think that you should not only test that it works, but also test what happens when it does not work.

i.e. you could stimulate no responses with this:
`cy.intercept("/rest/v1/suggestion*", [])`

File: src/pages/suggestions/index.tsx [hunk 2]
```diff
@@ -5,50 +5,70 @@ import { PlusIcon } from "@heroicons/react/24/solid"
...
+      )}
+
+      {noSuggestionsMade && (
+        <div className={"max-w-m flex items-center justify-center gap-4 text-zinc-400"}>
+          <div>
+            <MagnifyingGlassCircleIcon className={"h-[50px] w-[50px]"} />
+          </div>
+          <p>Looks like there are no suggestions made yet! Feel free to start adding them.</p>
+        </div>
+      )}
```
Comment (fkoulen) [inline src/pages/suggestions/index.tsx:71]: This is always shown when it is loading the suggestions. I think you want to hide it then as well.

File: cypress/e2e/suggestion.cy.ts (hunk 1 above)
Comment (fkoulen) [reply cypress/e2e/suggestion.cy.ts:4]: I've created an issue for this #199

File: cypress/e2e/suggestion.cy.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      cy.data("division").first().then((division) => {
+        const criteria = division.text().includes(username)
+        cy.data("division").first().click()
+        cy.data("division").first().should(`${criteria ? "not." : ""}contain.text`, username)
+      })
+    })
+
+  })
+
+  context("suggestion doesn't exists", () => {
```
Comment (fkoulen) [inline cypress/e2e/suggestion.cy.ts:26]: exist*

File: cypress/e2e/suggestions.cy.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  })
+
+  context("spinner", () => {
+    beforeEach(() => {
+      cy.login()
+      cy.visit("/suggestions")
+    })
+
+    it("should display spinner", () => {
```
Comment (fkoulen) [inline cypress/e2e/suggestions.cy.ts:57]: pretty vague name. 

`should display spinner when suggestions are still being retrieved` lets us know when it should display, now we don't necessary need to look at the code to see when it should display

**Review by rensdoodeman — APPROVED** [review]
